### ISSUE TITLE
aria-selected on 'row' element seems to be ignored when navigating through 'gridcell' elements of a 'grid'

### ISSUE THREAD

**Issue by turjmner8** [nvaccess/nvda#17626]
When using NVDA along with a 'grid' role element, the selection state of any 'row' role elements that are marked with aria-selected="true" seem to be ignored in virtual cursor navigation (using ctrl+alt arrows to navigate), and also when using real browser focus to navigate (regular arrow keys after pressing insert+space).

From the wai-aria spec, it seems like a selected row (as opposed to individually selected cells) should be represented with aria-selected="true" on the row element as it supports aria-selected. https://www.w3.org/TR/wai-aria-1.2/#row

When using JAWS 2025, the selection state is conveyed to the user when navigating in both, virtual cursor and real focus modes as I would expect.

### Steps to reproduce:

https://codepen.io/turjmner8/pen/gbYKrvy

Go to the above codepen, and navigate to the second data row (the one with Down Town Grocery description) using virtual cursor or real focus, and notice that no indication that the row is selected via aria-selected="true" attribute is given.

This sample is a copy of one of the W3C data grid examples, with this additional aria-selected markup added to the row. Doing the same in JAWS 2025 will convey the row selection state using both types of navigation.

### Actual behavior:

No indication of the aria-selected="true" is announced.

### Expected behavior:

Some indication of the row being selected should be announced.

**Comment by gerald-hartig** [comment]
Possibly related to #11986
